src/components/Header/SideNav/Dropdown.js

```diff
@@ -59,8 +60,17 @@ export default class Dropdown extends PureComponent {
   }
 
   render() {
-    const { items, children } = this.props
-    const { isExpanded } = this.state
+    const { items, children, path } = this.props
+    let { isExpanded } = this.state
+
+    const isServicePage = path.search(servicesRegExp) > -1
+    const serviceTitle = isServicePage ? path.match(servicesRegExp)[0] : null
+    const isSpecialityPage = path.includes('speciality')
+    const service = isSpecialityPage ? getService(path) : serviceTitle
+
+    isExpanded =
+      isExpanded ||
+      items.filter(item => item.to.startsWith(`/${service}`)).length > 0
```

Comment:
Could this lead to not being able to close a dropdown?

Reply:
ugh, yes, good catch.

Reply:
Also I didn't notice this yesterday but you're referencing state and changing its value in the render without setting the state of the actual component which is not great. React has component lifecycle methods for what we're trying to do here